lib/phobos_db_checkpoint/handler.rb

```diff
@@ -31,7 +45,15 @@ def around_consume(payload, metadata)
           end
 
           event_action = instrument('db_checkpoint.event_action', event_metadata) do
-            yield
+            begin
```

Comment:
Your suggestion basically boils down to some derivative of this sample code below:

```ruby
      def retry_consume?(event, event_metadata, exception)
        return false unless Phobos.config&.db_checkpoint&.max_retries
        event_metadata[:retry_count] < Phobos.config&.db_checkpoint&.max_retries
      end
```

Your suggestion is that if max_retries isn't set at all or set to nil, then the expected result is keep retrying forever...

However, is that very useful outside of implementations using Phobos that handles errors themselves? 
Would a vanilla Phobos+PhobosDBCheckpoint user expect an event to "get stuck" and fail in a never ending loop?

From this point of view I must say I am more strongly in favor of that if you haven't configured a max retry it should fail immediately instead.